ldclient/client.py

```diff
@@ -0,0 +1,229 @@
+                 consumer_class=None):
+        if defaults is None:
+            defaults = {}
+
+        self.base_uri = base_uri.rstrip('\\')
+        self.stream_uri = stream_uri.rstrip('\\')
+        self.stream = stream
+        self.stream_processor_class = RequestsStreamProcessor if not stream_processor_class else stream_processor_class
+        self.feature_store_class = InMemoryFeatureStore if not feature_store_class else feature_store_class
+        self.consumer_class = RequestsEventConsumer if not consumer_class else consumer_class
+        self.feature_requester_class = RequestsFeatureRequester if not feature_requester_class else \
+            feature_requester_class
+        self.connect = connect_timeout
+        self.read = read_timeout
+        self.upload_limit = upload_limit
+        self.capacity = capacity
+        self.verify = verify
+        self.defaults = defaults
+
+    def get_default(self, key, default):
+        return default if key not in self.defaults else self.defaults[key]
+
+    @classmethod
+    def default(cls):
+        return cls()
+
+
+class InMemoryFeatureStore(FeatureStore):
+    def __init__(self):
+        self._lock = ReadWriteLock()
+        self._initialized = False
+        self._features = {}
+
+    def get(self, key):
+        try:
+            self._lock.rlock()
+            f = self._features.get(key)
+            if f is None or 'deleted' in f and f['deleted']:
+                return None
+            return f
+        finally:
+            self._lock.runlock()
+
+    def all(self):
+        try:
+            self._lock.rlock()
+            return dict((k, f) for k, f in self._features.items() if ('deleted' not in f) or not f['deleted'])
+        finally:
+            self._lock.runlock()
+
+    def init(self, features):
+        try:
+            self._lock.lock()
+            self._features = dict(features)
+            self._initialized = True
+        finally:
+            self._lock.unlock()
+
+    # noinspection PyShadowingNames
+    def delete(self, key, version):
+        try:
+            self._lock.lock()
+            f = self._features.get(key)
+            if f is not None and f['version'] < version:
+                f['deleted'] = True
+                f['version'] = version
+            elif f is None:
+                f = {'deleted': True, 'version': version}
+                self._features[key] = f
+        finally:
+            self._lock.unlock()
+
+    def upsert(self, key, feature):
+        try:
+            self._lock.lock()
+            f = self._features.get(key)
+            if f is None or f['version'] < feature['version']:
+                self._features[key] = f
+        finally:
+            self._lock.unlock()
+
+    @property
+    def initialized(self):
+        try:
+            self._lock.rlock()
+            return self._initialized
+        finally:
+            self._lock.runlock()
+
+
+class LDClient(object):
```

Comment:
I preferred having `TwistedLDClient`. Most of our customers don't need to set additional config options, so being able to instantiate a client in one line via:

```
client = TwistedLDClient("API_KEY")
```

is nice.